src/lib/shell/commands/Config.cpp

```diff
@@ -96,10 +96,32 @@ static CHIP_ERROR ConfigGetSetupPinCode(bool printHeader)
     {
         streamer_printf(sout, "PinCode:         ");
     }
-    streamer_printf(sout, "%09u\r\n", setupPinCode);
+    streamer_printf(sout, "%08u\r\n", setupPinCode);
     return CHIP_NO_ERROR;
 }
 
+static CHIP_ERROR ConfigSetSetupPinCode(char * argv)
```

Comment:
The device should not need to know the "PIN code" (i.e. commissioning passcode), just the w0/L of the verifier. We use PIN today because we did not design an easy way to have the onboarding payload separate from running binary, but it is problematic since it forces embedded target to compute the verifier itself, which in turn forces us to use an iteration count that is too small for PBKDF.

Obviously it would be be disruptive to remove the passcode in the short term, but adding more new commands that enshrine usage that doesn't match to spec intent make the transition to the final approach more problematic

Reply:
Ok, I got your point, then would you be ok with adding a command only to set discriminator? That should be enough to clearly distinguish devices that we want to commission, but would not introduce further deviations from spec.

Reply:
Yes, definitely you can keep the discriminator setter and I wouldn't be against having a short term passcode setter until we sort it all out, but this is an area where we have to start to move towards more realistic usage and APIs for accessing the data. Will open an issue for discussion since my Slack thread did not get traction

Reply:
So what is the conclusion, should I remove setter for passcode now, or we want to leave it and create issue, as a reminder to do that? Sorry, but after your approval I a little bit got lost and I don't know if I can to resolve conversation?